### ISSUE TITLE
Incorrect handling of Content-Length after HEAD responses

### ISSUE THREAD

**Issue by Lukasa** [nodejs/llhttp#202]
Returning 1 from `on_headers_complete` signals to llhttp that the message should be assumed to have no body, and the parser should reset. This is supposed to be used for responses to HEAD requests.

Responses to HEAD requests may have Content-Length headers to match the expected response length. llhttp parses these, but when it receives 1 as the return code from `on_headers_complete`, it does not clear the `content_length` value. As a result, the next response received using that parser will have an incorrect content length caused by lexicographically prepending the previous content length to the new one.

For example, the following transaction:

```
=> HEAD / HTTP/1.1\r\nHost: example.com\r\n\r\n
<= 200 OK\r\nContent-Length: 123\r\n\r\n
=> GET / HTTP/1.1\r\nHost: example.com\r\n\r\n
<= 200 OK\r\nContent-Length: 456\r\n\r\n
```

Will cause the parser to believe that the content length of the second message is 123456 bytes. This will lead to it incorrectly parsing the subsequent responses.

**Comment by pallas** [comment]
Are you calling reset?

**Comment by Lukasa** [comment]
No, historically there's been no need to: llhttp can happily parse a second message after the first without a call to reset. This is the only case that behaves differently.

**Comment by Lukasa** [comment]
Incidentally, `reset` also doesn't reset the content_length field:

https://github.com/nodejs/llhttp/blob/115ad728005deafb3e41b22aae0ff1c04820d35e/src/native/api.c#L125-L137

**Comment by Lukasa** [comment]
Relatedly, node itself never calls `reset`.
